velox/exec/tests/PrefixSortTest.cpp

```diff
@@ -0,0 +1,355 @@
+    }
+    return fmt::format(
+        "sortCompareFlags:{}, \nexpectedResult:\n{}",
+        sortCompareFlagsStr.str(),
+        expectedResultStr);
+  }
+
+ private:
+  std::vector<CompareFlags> sortCompareFlags_;
+  RowVectorPtr expectedResult_;
+};
+
+class PrefixSortTest : public OperatorTestBase {
+ protected:
+  std::vector<char*>
+  storeRows(int numRows, const RowVectorPtr& sortedRows, RowContainer* data);
+
+  void testSort(
+      int numRows,
+      const std::vector<TestSetting> testSettings,
+      const RowVectorPtr& sortedRows);
+
+  // Use std::sort to generate expected result.
+  const RowVectorPtr generateExpectedResult(
+      const std::vector<CompareFlags> compareFlags,
+      int numRows,
+      const RowVectorPtr& sortedRows);
+
+  // 20 MB
+  const int64_t maxBytes_ = 20LL << 20;
+  const std::shared_ptr<memory::MemoryPool> rootPool_{
+      memory::MemoryManager::getInstance()->addRootPool(
+          "PrefixSortTest",
+          maxBytes_)};
+
+  const std::shared_ptr<memory::MemoryPool> pool_{
+      rootPool_->addLeafChild("PrefixSortTest", maxBytes_)};
+
+  const std::shared_ptr<memory::MemoryPool> fuzzerPool =
+      rootPool_->addLeafChild("VectorFuzzer");
+};
+
+std::vector<char*> PrefixSortTest::storeRows(
+    int numRows,
+    const RowVectorPtr& sortedRows,
+    RowContainer* data) {
+  std::vector<char*> rows;
+  SelectivityVector allRows(numRows);
+  rows.resize(numRows);
+  for (int row = 0; row < numRows; ++row) {
+    rows[row] = data->newRow();
+  }
+  for (int column = 0; column < sortedRows->childrenSize(); ++column) {
+    DecodedVector decoded(*sortedRows->childAt(column), allRows);
+    for (int i = 0; i < numRows; ++i) {
+      char* row = rows[i];
+      data->store(decoded, i, row, column);
+    }
+  }
+  return rows;
+}
+
+const RowVectorPtr PrefixSortTest::generateExpectedResult(
+    const std::vector<CompareFlags> compareFlags,
+    int numRows,
+    const RowVectorPtr& sortedRows) {
+  const auto sortedRowsType = asRowType(sortedRows->type());
+  const int sortKeysNum = sortedRows->childrenSize();
+  RowContainer rowContainer(sortedRowsType->children(), pool_.get());
+
+  std::vector<char*> rows = storeRows(numRows, sortedRows, &rowContainer);
+  std::sort(
+      rows.begin(), rows.end(), [&](const char* leftRow, const char* rightRow) {
+        for (vector_size_t index = 0; index < sortKeysNum; ++index) {
+          if (auto result = rowContainer.compare(
+                  leftRow, rightRow, index, compareFlags[index])) {
+            return result < 0;
+          }
+        }
+        return false;
+      });
+  const RowVectorPtr result =
+      BaseVector::create<RowVector>(sortedRowsType, numRows, pool_.get());
+  for (int column = 0; column < compareFlags.size(); ++column) {
+    rowContainer.extractColumn(
+        rows.data(), numRows, column, result->childAt(column));
+  }
+  return result;
+}
+
+void PrefixSortTest::testSort(
```

Comment:
This function is buggy. It doesn't take into account that sortedRows may have payload columns (as in fuzzMulti). I'm fixing this.